django/contrib/gis/gdal/raster/source.py

```diff
@@ -443,6 +462,9 @@ def transform(self, srs, driver=None, name=None, resampling='NearestNeighbour',
                 'Transform only accepts SpatialReference, string, and integer '
                 'objects.'
             )
+
+        if target_srs.srid == self.srid:
+            return self.clone()
```

Comment:
The name provided in the "transform" call could be passed to the clone function to create the clone in the expected location.

Reply:
Great catch :+1: